infrastructure/sync.sh

```diff
@@ -6,12 +6,35 @@ site_archive="$1"
 # S3 URL of the destination website bucket (e.g., the one hosting www.pulumi.com).
 site_bucket="$2"
 
+if [ -z "$site_archive" ]; then
+    echo "Missing argument: site_archive"
+    exit 1
+fi
+
+if [ -z "$site_bucket" ]; then
+    echo "Missing argument: site_bucket"
+    exit 1
+fi
+
 # Download the archive from the archive bucket and unpack it into a local folder.
 echo "Downloading $site_archive..."
 aws s3 cp $site_archive .
 mkdir site_contents
 tar -xzvf $(basename "$site_archive") -C site_contents
 
+# Perform a couple of sanity checks to make sure the unpacked archive looks like a proper
+# website docroot (e.g., one composed of at least a thousand index.html files, including
+# one at the root).
+if [ -z "$(ls -al site_contents/index.html)" ]; then
```

Comment:
Rather than checking for the output of `ls` (which, IDK, maybe could just print out with "file not found"), wouldn't it make more sense to use the `-f` flag here? Which I believe texts specifically if a file exists or not?

`if [ ! -f "./site_contents/index.html" ]; then` seems like a better approach if I have that right.